modules/ort/src/main/kotlin/workflow/processors/enricher/OrtDownloaderProcessor.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.ort.workflow.processors.enricher
+
+import com.here.ort.downloader.DownloadException
+import com.here.ort.downloader.Downloader
+import com.here.ort.utils.encodeOrUnknown
+import com.here.ort.utils.packZip
+import org.eclipse.sw360.antenna.model.artifact.Artifact
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceFile
+import org.slf4j.LoggerFactory
+import org.eclipse.sw360.antenna.ort.utils.ArtifactToPackageMapper
+import java.io.File
+import org.eclipse.sw360.antenna.api.workflow.AbstractProcessor
+import org.eclipse.sw360.antenna.model.artifact.ArtifactCoordinates
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename
+
+class OrtDownloaderProcessor : AbstractProcessor() {
+    companion object {
+        const val ORT_DOWNLOADER_DIR = "ort-downloader-result"
+        private val LOGGER = LoggerFactory.getLogger(OrtDownloaderProcessor::class.java)
+    }
+
+    init {
+        workflowStepOrder = 350
+    }
+
+    private lateinit var ortDownloadFolder: File
+
+    override fun configure(configMap: MutableMap<String, String>) {
+        ortDownloadFolder = context.toolConfiguration.antennaTargetDirectory.resolve(ORT_DOWNLOADER_DIR).toFile()
+        ortDownloadFolder.takeUnless { it.exists() }?.let { it.mkdirs() }
```

Comment:
`exists()` is too unspecific here, it should better be `isDirectory`. Also, I find the logic a bit hard to follow as you're not using the non-null return value of `takeUnless` at all. How about simply:

    if (!ortDownloadFolder.isDirectory) ortDownloadFolder.mkdirs()

Or, yet better as you check also for an error (e.g. is already a file of the name name exists):

    if (!ortDownloadFolder.isDirectory && !ortDownloadFolder.mkdirs()) throw ...